src/dialect/mod.rs

```diff
@@ -561,6 +561,11 @@ pub trait Dialect: Debug + Any {
     fn supports_asc_desc_in_column_definition(&self) -> bool {
         false
     }
+
+    /// For example: SELECT col_alias = col FROM tbl
```

Comment:
```suggestion
    /// Returns true if this dialect supports treating the equals operator `=` within a [`SelectItem`]
    /// as an alias assignment operator, rather than a boolean expression.
    /// For example: the following statements are equivalent for such a dialect:
    /// ```sql
    ///  SELECT col_alias = col FROM tbl;
    ///  SELECT col_alias AS col FROM tbl;
    /// ```
```
Something like this to illustrate what the flag implies Im thinking?